Common/Securities/Option/OptionFilterUniverse.cs

```diff
@@ -224,6 +225,496 @@ public OptionFilterUniverse PutsOnly()
+                (x.ID.OptionRight == OptionRight.Put && x.ID.StrikePrice == nearPutStrike) ||
+                (x.ID.OptionRight == OptionRight.Call && x.ID.StrikePrice == farCallStrike) ||
+                (x.ID.OptionRight == OptionRight.Put && x.ID.StrikePrice == farPutStrike)
+            ));
+            if (filtered.Count() != 4)
+            {
+                return this.WhereContains( new List<Symbol>() );
+            }
+            return filtered;
+        }
+
+        /// <summary>
+        /// Sets universe of an OTM call, an ITM call, an OTM put, and an ITM put with the same expiry with closest match to the criteria given.
+        /// The OTM call has the same strike as the ITM put, while the same holds for the ITM call and the OTM put
+        /// </summary>
+        /// <param name="minDaysTillExpiry">The minimum days till expiry from the current time, closest expiry will be selected</param>
+        /// <param name="strikeSpread">The desire strike price distance of the OTM call and the OTM put from the current underlying price</param>
+        /// <remarks>Applicable to Long and Short Box Spread Option Strategy</remarks>
+        /// <returns>Universe with filter applied</returns>
+        public OptionFilterUniverse BoxSpread(int minDaysTillExpiry = 30, decimal strikeSpread = 5)
+        {
+            if (strikeSpread <= 0)
+            {
+                throw new ArgumentException($"BoxSpread(): strike arguments must be positive, {nameof(strikeSpread)}");
+            }
+
+            // Select the expiry as the nearest to set days later
+            var expiry = GetExpiry(AllSymbols, minDaysTillExpiry);
+            var contracts = AllSymbols.Where(x => x.ID.Date == expiry).ToList();
+            if (contracts.Count == 0)
+            {
+                return this.WhereContains( new List<Symbol>() );
+            }
+
+            // Select the strike prices with the set spread range
+            var higherStrike = GetStrike(contracts.Where(x => x.ID.StrikePrice > Underlying.Price), strikeSpread);
+            var lowerStrike = GetStrike(contracts.Where(x => x.ID.StrikePrice < higherStrike && x.ID.StrikePrice < Underlying.Price), -strikeSpread);
+
+            // Select the contracts
+            var filtered = this.Where(x => 
+                (x.ID.StrikePrice == higherStrike || x.ID.StrikePrice == lowerStrike) &&
+                x.ID.Date == expiry);
+            if (filtered.Count() != 4)
+            {
+                return this.WhereContains(new List<Symbol>());
+            }
+            return filtered;
+        }
+
+        private DateTime GetExpiry(IEnumerable<Symbol> symbols, int minDaysTillExpiry)
+        {
+            var leastExpiryAccepted = _lastExchangeDate.AddDays(minDaysTillExpiry);
+            var accepted = symbols.Where(x => x.ID.Date >= leastExpiryAccepted);
+            if (!accepted.Any())
+            {
+                return DateTime.MaxValue;
+            }
```

Comment:
Let's just remove this and perform a FIrstOrDefault bellow to avoid enumerating twice